### PR TITLE
Replacing parameter_dictionary with self.parameters

### PR DESCRIPTION
**Pull request by krishnakruthi**

### CODE REVIEW

**Comment by adivijaykumar** [comment]
@Kruthi24  to start off --- you should not have the .DS_Store files in the branch at all. Don't stage them. Could you please delete them?

File: .gitignore [hunk 1]
```diff
@@ -15,3 +15,4 @@ MANIFEST
 *.ipynb_checkpoints
 outdir/*
 .idea/*
+.DS_Store
```
Comment (adivijaykumar) [inline .gitignore:18]: This is fine for now but we will have to remove this before merging into the main repo.

**Comment by adivijaykumar** [comment]
@Kruthi24 do you want to merge this right now or do you want to add more changes?

**Comment by krishnakruthi** [comment]
I'll add more changes.

**Pull request merged**